otherlibs/systhreads/st_stubs.c

```diff
@@ -276,18 +270,15 @@ static caml_thread_t caml_thread_new_info(void)
 
 static value caml_thread_new_descriptor(value clos)
 {
-  value mu = Val_unit;
+  CAMLparam1(clos);
+  CAMLlocal1(mu);
   value descr;
-  Begin_roots2 (clos, mu)
-    /* Create and initialize the termination semaphore */
-    mu = caml_threadstatus_new();
-    /* Create a descriptor for the new thread */
-    descr = caml_alloc_small(3, 0);
-    Ident(descr) = Val_long(atomic_fetch_add(&thread_next_id, +1));
-    Start_closure(descr) = clos;
-    Terminated(descr) = mu;
-  End_roots();
-  return descr;
+  /* Create and initialize the termination semaphore */
+  mu = caml_threadstatus_new();
+  /* Create a descriptor for the new thread */
+  descr = caml_alloc_3(3, Val_long(atomic_fetch_add(&thread_next_id, +1)),
```

Comment:
Same here, the tag used to be `0` and now its `3`.

Reply:
Thanks, see #11086